templates/CONTRIBUTING_GUIDE.md

```diff
@@ -0,0 +1,121 @@
+# Contributing
+
+This package is maintained by the [TEAM] team. Ask for help or a review in the [#[TEAM SLACK NAME]]([TEAM SLACK NAME LINK]) Slack channel.
+
+## :gear: Setup
+
+<!-- When applicable, list system requriements to work on the project. -->
+
+The following setup is required to work on this project:
+
+- Prerequisite 1
+- Prerequisite 2
+- Prerequisite 3
+
+## :memo: Project-specific notes
+
+<!-- Share information about the repository. -->
+<!-- What specific knowledge do contributors need? -->
+
+> [!TIP]
+> Please update this section with helpful notes for contributors.
+
+## :construction_worker: Develop
+
+> [!NOTE]
+> It's highly recommended to discuss your changes with the [TEAM] team before starting.
+>
+> A short discussion can accellerate your work and ship it faster.
+
+```sh
+# Clone and prepare the project.
+git clone [REPO SSH ADRESS]
+cd [REPO NAME]
+[DEPENDENCY INSTALL COMMAND]
+
+# Create a new branch for your changes (e.g. lh/fix-win32-paths).
+git checkout -b <your-initials>/<feature-or-fix-description>
+
+# Start the development server.
+# Run this command while you are working on your changes.
+[DEV SERVER COMMAND]
+
+# Build the project for production.
+# Run this command when you want to see the production version.
+[BUILD COMMAND]
+
+# Lint your changes before requesting a review. No errors are allowed.
+[LINT COMMAND]
+
+# Format your changes before requesting a review. No errors are allowed.
+[FORMAT COMMAND]
+
+# Test your changes before requesting a review.
+# All changes should be tested. No failing tests are allowed.
+[TEST COMMAND]
+```
+
+## :building_construction: Submit a pull request
+
+> [!NOTE]
+> Code should be reviewed by the [TEAM] team before merging.
+>
+> Request a review in the [#[TEAM SLACK NAME]]([TEAM SLACK LINK]) Slack channel.
+
+```sh
+# Open a pull request. This example uses the GitHub CLI.
+gh pr create
+
+# Perform a review within your team. This review should at least consider
+# the PR's general direction, code style, and test coverage.
+
+# Once approved, request a review from the [TEAM] team in #[TEAM SLACK NAME].
+# This second review should verify the pull request is not missing crucial
+# logic.
+
+# When ready, PRs should be merged using the "Squash and merge" option.
+```
+
+<!-- FOR SERVICES: -->
+<!-- Delete this section if not applicable. -->
+
+## :rocket: Deploy
+
+Deployments are broken into three stages: development, staging, and production.
+
+#### Development environment
+
+<!-- Describe how to deploy this service to the development environment. -->
+
+#### Staging
+
+<!-- Describe how to deploy this service to the staging environment. -->
+
+#### Production
```

Comment:
Should we specify that
- only the main branch of services is to be deployed
- it's people's responsibility to make sure their PR doesn't break production or other services (by running CI, creating tests, ...)

Reply:
I don't know if a rollback strategy make sense to add here but. could be a good addition

Reply:
> Should we specify that only the main branch of services is to be deployed

I think this info is specific to each service, so we can probably leave it out of the template. It can be mentioned in the "Production" section.

> Should we specify that it's people's responsibility to make sure their PR doesn't break production or other services (by running CI, creating tests, ...)

Yes, that is a good addition, similar to the "no failing tests are allowed" line in the "Develop" section. It could be added as a [Caution block](https://github.com/orgs/community/discussions/16925). immediately after the "Production" heading.

> I don't know if a rollback strategy make sense to add here but. could be a good addition

Makes sense to me as part of the Production section.

This guide should be short and concise while providing enough information to get started. The last thing someone wants to do when they start working is read a bunch of documentation. 😅 Do you have an idea of how we could do that?

Reply:
> Should we specify that only the main branch of services is to be deployed

I think this info is specific to each service, so we can probably leave it out of the template. It can be mentioned in the "Production" section.

> Should we specify that it's people's responsibility to make sure their PR doesn't break production or other services (by running CI, creating tests, ...)

Yes, that is a good addition, similar to the "no failing tests are allowed" line in the "Develop" section. It could be added as a [Caution block](https://github.com/orgs/community/discussions/16925). immediately after the "Production" heading.

> I don't know if a rollback strategy make sense to add here but. could be a good addition

Makes sense to me as part of the Production section.

This guide should be short and concise while providing enough information to get started. The last thing someone wants to do when they start working is read a bunch of documentation. 😅 Do you have an idea of how we could do that?